packages/sdk/src/Portfolio/Integrations/Curve.ts

```diff
@@ -233,18 +230,36 @@ export type RedeemArgs = {
   pool: Address;
   outgoingLpTokenAmount: bigint;
   useUnderlyings: boolean;
-  redeemType: RedeemType;
-  incomingAssetsData: Hex;
-};
+} & (StandardRedeemArgs | OneCoinRedeemArgs);
 
 export function redeemEncode(args: RedeemArgs): Hex {
-  return encodeAbiParameters(redeemEncoding, [
-    args.pool,
-    args.outgoingLpTokenAmount,
-    args.useUnderlyings,
-    args.redeemType,
-    args.incomingAssetsData,
-  ]);
+  const redeemType = args.redeemType;
+
+  switch (redeemType) {
+    case RedeemType.Standard: {
+      const incomingAssetsData = standardRedeemEncode(args.orderedMinIncomingAssetAmounts);
+
+      return encodeAbiParameters(redeemEncoding, [
+        args.pool,
+        args.outgoingLpTokenAmount,
+        args.useUnderlyings,
+        redeemType,
+        incomingAssetsData,
+      ]);
+    }
+
+    case RedeemType.OneCoin: {
+      const incomingAssetsData = oneCoinRedeemEncode(args.incomingAssetPoolIndex, args.minIncomingAssetAmount);
+
+      return encodeAbiParameters(redeemEncoding, [
+        args.pool,
+        args.outgoingLpTokenAmount,
+        args.useUnderlyings,
+        args.redeemType,
+        incomingAssetsData,
+      ]);
+    }
```

Comment:
maybe we could assertNever at the end of the switch. Same comment to all of the switch statements in the PR